src/sync.js

```diff
@@ -419,27 +419,26 @@ SyncApi.prototype.sync = function() {
     //      them from /sync.
     //   2) We need to get/create a filter which we can use for /sync.
 
-    function getPushRules() {
-        client.getPushRules().done((result) => {
+    async function getPushRules() {
+        try {
+            const result = await client.getPushRules();
             debuglog("Got push rules");
 
             client.pushRules = result;
-
-            getFilter(); // Now get the filter and start syncing
-        }, (err) => {
+        } catch (err) {
             // wait for saved sync to complete before doing anything else,
             // otherwise the sync state will end up being incorrect
-            savedSyncPromise.then(() => {
-                const keepaliveProm = self._startKeepAlives();
-                self._updateSyncState("ERROR", { error: err });
-                return keepaliveProm;
-            }).then(() => {
-                getPushRules();
-            });
+            await savedSyncPromise;
+            const keepaliveProm = self._startKeepAlives();
+            self._updateSyncState("ERROR", { error: err });
+            await keepaliveProm;
```

Comment:
The 6 lines above should probably be factored out from here and from getPushRules

Reply:
done, although suggestions welcome on function name